src/chainstate/burn/db/sortdb.rs

```diff
@@ -1585,6 +1603,66 @@ impl<'a> SortitionHandleConn<'a> {
         })
     }
 
+    /// is the given block a descendant of `potential_ancestor`?
+    ///  * block_at_burn_height: the burn height of the sortition that chose the stacks block to check
+    ///  * potential_ancestor: the stacks block hash of the potential ancestor
+    pub fn descended_from(
```

Comment:
Is there any way to DRY this up?  Looks like it was copy/pasted from `SortitionHandleTx<'_>`.

Reply:
I can try. I didn't want to DRY it too much initially, because any approach to doing so may end obfuscating the function quite a bit.

Reply:
One thing we could do is implement a trait for both `SortitionHandleTx<'_>` and `SortitionHandleConn<'_>` that exposes all the internal state we care about in order to execute this.  Like, `self.get_block_snapshot_by_height()` and `self.tx()` could be the same for both of these (but instead of `fn tx(&mut self)`, you'd have `fn conn(&self)` to return an immutable reference to the inner sqlite connection).

Reply:
Makes sense. Added in 9e676926c

Reply:
So is this instance of `descended_from()` redundant, now that `SortitionHandle::descended_from()` exists?

Reply:
Yep, removed in 9d00a6cfd48fe557dc524a6935e7ab9c1af74d79